components/automate-cli/cmd/chef-automate/certRotate.go

```diff
@@ -483,21 +519,46 @@ func patchOSNodeDN(flagsObj *certRotateFlags, patchFnParam *patchFnParameters, c
 	return nil
 }
 
-func getOldCn(automatesConfig map[string]*deployment.AutomateConfig) string {
+func (c *certRotateFlow) getFrontIpsToSkipRootCAandCNPatchingForOs(automatesConfig map[string]*deployment.AutomateConfig, newRootCA string, newCn string, node string, infra *AutomateHAInfraDetails) []string {
+	oldRootCA := ""
 	oldCn := ""
-	for _, config := range automatesConfig {
+	skipIpsList := []string{}
+	for ip, config := range automatesConfig {
 		if config.Global.V1.External.Opensearch != nil && config.Global.V1.External.Opensearch.Ssl != nil {
 			if config.Global.V1.External.Opensearch.Ssl.ServerName != nil {
 				oldCn = config.Global.V1.External.Opensearch.Ssl.ServerName.Value
-				break
+				if c.getFrontEndIpsForSkippingCnAndRootCaPatching(newRootCA, newCn, oldCn, oldRootCA, node) {
+					skipIpsList = append(skipIpsList, ip)
+				}
+			}
+			if config.Global.V1.External.Opensearch.Ssl.RootCert != nil {
+				oldRootCA = config.Global.V1.External.Opensearch.Ssl.RootCert.Value
+				if c.getFrontEndIpsForSkippingCnAndRootCaPatching(newRootCA, newCn, oldCn, oldRootCA, node) {
+					skipIpsList = append(skipIpsList, ip)
+				}
 			}
 		}
 	}
-	return oldCn
+	return skipIpsList
+}
+
+// getFrontIpsToSkipRootCAPatchingForPg compare new root-ca and current root-ca of remoteService and returns ips to skip root-ca patching.
+func (c *certRotateFlow) getFrontIpsToSkipRootCAPatchingForPg(automatesConfig map[string]*deployment.AutomateConfig, newRootCA string, infra *AutomateHAInfraDetails) []string {
+	oldRootCA := ""
+	skipIpsList := []string{}
+	for ip, config := range automatesConfig {
+			if config.Global.V1.External.Postgresql.Ssl.RootCert != nil {
+				oldRootCA = config.Global.V1.External.Postgresql.Ssl.RootCert.Value
+				if oldRootCA == newRootCA {
+					skipIpsList = append(skipIpsList, ip)
+				}
+		}
+	}
+	return skipIpsList
 }
 
 // patchConfig will patch the configurations to required nodes.
-func (c *certRotateFlow) patchConfig(param *patchFnParameters) error {
+func (c *certRotateFlow) patchConfig(param *patchFnParameters, concurrent bool) error {
```

Comment:
Can we add the `concurrent` as one of the fields in the `patchFnParameters` struct?

Reply:
done